WORKSPACE

```diff
@@ -119,6 +119,30 @@ go_repository(
     version = "v0.6.0",
 )
 
+go_repository(
+    name = "googleapis",
+    build_directives = [
+        "gazelle:go_generate_proto false",
```

Comment:
If we expect users to add these directives, we may want to add them to https://github.com/bazelbuild/bazel-gazelle/blob/master/internal/bzlmod/default_gazelle_overrides.bzl so that they don't have to. This would also soften the migration.

Reply:
Since `googleapis` is not `go_deps`, [default_gazelle_overrides.bzl](https://github.com/bazelbuild/bazel-gazelle/blob/master/internal/bzlmod/default_gazelle_overrides.bzl) won't make a difference, right?

Reply:
That's true.

Do we happen to have an example for how depending on this would look like with Bzlmod? I am a bit worried that strict repo visibility would make it difficult for users to specify the correct resolve directives.

Reply:
Is it possible to load a `go_repository` in bzlmod? Worse case, we can pre-generate all the `proto_library` targets and upload it to BCR.

Reply:
It is, not if we do this in the most direct way possible, users won't easily be able to update the version, bringing us back to the problem we are trying to solve.

I will think more about this and get back to you.

Reply:
We can also commit all missing `proto_library` rules to googleapis repo, which is already a bazel repo. It gives us a bit more protection because their CI may catch some issues when the proto files and build rules are out of sync.

Reply:
I think that the best model would be a `googleapi` Bazel module that only contains the `proto_library` targets and `.proto` files and separate `googleapi-lang` repos with only the `lang_proto_library` targets.

Reply:
The original googleapis repo contains both `proto_library` (although not for every proto file, which is why we need to generate more) and `lang_proto_library` targets. I think it's a good idea to use it as a pure proto Bazel module.

I am not sure whether we need another googleapi-lang repos though. For Go, all generated packages have import prefix like google.golang.org/genproto/googleapis. They have a checked-in version published to that path. The plan is to use the go_deps `org_golang_google_genproto` instead. WDYT?

Reply:
With https://github.com/googleapis/googleapis/pull/831, we can use http_archive for googleapis